experimental/internal/taxa/names.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taxa
+
+var (
+	// names is an array of user-visible names of all of the productions in this
+	// package.
+	names = [...]string{
+		Unknown:      "<unknown>",
+		Unrecognized: "unrecognized token",
+		TopLevel:     "file scope",
+		EOF:          "end-of-file",
+
+		Decl:         "declaration",
+		Empty:        "empty declaration",
+		Syntax:       "`syntax` declaration",
+		Edition:      "`edition` declaration",
+		Package:      "`package` declaration",
+		Import:       "import",
+		WeakImport:   "weak import",
+		PublicImport: "public import",
+		Extensions:   "extension range",
+		Reserved:     "reserved range",
+		Body:         "definition body",
+
+		Def:     "definition",
+		Message: "message definition",
+		Enum:    "enum definition",
+		Service: "service definition",
+		Extend:  "message extension block",
+		Oneof:   "oneof definition",
+
+		Option:       "option setting",
+		CustomOption: "custom option setting",
+
+		Field:     "message field",
+		EnumValue: "enum value",
+		Method:    "service method",
+
+		CompactOptions: "compact options",
+		MethodIns:      "method parameter list",
+		MethodOuts:     "method return type",
+
+		FieldTag:    "message field tag",
+		OptionValue: "option setting value",
+
+		Path:            "path",
+		ExtensionInPath: "path in `(...)`",
+
+		Expr:      "expression",
+		Range:     "range expression",
+		Array:     "array expression",
+		Dict:      "dictionary expression",
+		DictField: "key-value pair",
```

Comment:
I worry that these are also not that intuitive for users -- Protobuf doesn't have dictionary types. It would likely be much more comprehensible to users to say "message literal" and "message field". I know that's complicated by the fact that we may want to use these AST nodes/productions in other contexts (outside of message literals and fields) in the future.